### PR TITLE
Enable push and build to sonatype

### PR DESCRIPTION
**Pull request by Diablo2050**

### CODE REVIEW

File: .bsp/sbt.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"name":"sbt","version":"1.5.2","bspVersion":"2.0.0-M5","languages":["scala"],"argv":["/usr/lib/jvm/java-11-openjdk-amd64/bin/java","-Xms100m","-Xmx100m","-classpath","/home/amr/.cache/sbt/boot/sbt-launch/1.5.2/sbt-launch-1.5.2.jar","xsbt.boot.Boot","-bsp","--sbt-launch-jar=/home/amr/.cache/sbt/boot/sbt-launch/1.5.2/sbt-launch-1.5.2.jar"]}
```
Comment (simplesteph) [inline .bsp/sbt.json:1]: is that file needed?

File: .github/workflows/scala.yml [hunk 2]
```diff
@@ -25,6 +25,22 @@ jobs:
       run: sbt universal:packageBin
       if: startsWith(github.ref, 'refs/tags/v')
 
+    # - name: publish
+    #   run: sbt publishSigned
```
Comment (simplesteph) [inline .github/workflows/scala.yml:29]: lots of commented stuff.. is that needed? or can we remove?

File: .github/workflows/scala.yml [hunk 3]
```diff
@@ -25,6 +25,22 @@ jobs:
...
+      env:
+        PGP_PASSPHRASE: ${{ secrets.PGP_PASSPHRASE }}
+        PGP_SECRET: ${{ secrets.PGP_SECRET }}
+        SONATYPE_PASSWORD: ${{ secrets.SONATYPE_PASSWORD }}
+        SONATYPE_USERNAME: ${{ secrets.SONATYPE_USERNAME }}
+    # - name: publish
+    #   run: sbt publishSigned
+
+    # - name: release snapshot
+    #   run: sonatypeRelease
```
Comment (simplesteph) [inline .github/workflows/scala.yml:42]: same for these comments

File: build.sbt [hunk 4]
```diff
@@ -1,6 +1,21 @@
+inThisBuild(List(
+  organization := "io.github.conduktor",
```
Comment (simplesteph) [inline build.sbt:2]: why had the organization changed?
GitHub's domain is `github.com`, not? 
Why did you remove the `.ksm`?

File: build.sbt [hunk 5]
```diff
@@ -1,6 +1,21 @@
...
+  homepage := Some(url("https://github.com/olafurpg/sbt-ci-release")),
```
Comment (simplesteph) [inline build.sbt:3]: here, it should be the github URL of this repo, no?

File: build.sbt [hunk 6]
```diff
@@ -1,6 +1,21 @@
...
+      url("https://github.com/conduktor")
+    )
+  )
+))
+
+
 name := "kafka-security-manager"
 
-organization := "com.github.conduktor.ksm"
+// organization := "io.github.conduktor"
```
Comment (simplesteph) [inline build.sbt:18]: remove then if commented?

File: build.sbt [hunk 7]
```diff
@@ -61,7 +76,12 @@ dockerUpdateLatest := true
 dockerBaseImage := "openjdk:8-jre-slim"
 
 // Add the default sonatype repository setting
-publishTo := sonatypePublishTo.value
+// publishTo := sonatypePublishTo.value
+// publishTo := sonatypePublishToBundle.value
+// ThisBuild / sonatypeCredentialHost := "s01.oss.sonatype.org"
+// sonatypeCredentialHost := "s01.oss.sonatype.org"
+// sonatypeRepository := "https://s01.oss.sonatype.org/service/local"
```
Comment (simplesteph) [inline build.sbt:83]: remove if commented?

File: publish.sbt [hunk 8]
```diff
@@ -0,0 +1,31 @@
+ThisBuild / organization := "io.github.conduktor"
```
Comment (simplesteph) [inline publish.sbt:1]: same comments about organization name here

File: publish.sbt [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+ThisBuild / scmInfo := Some(
+  ScmInfo(
+    url("https://github.com/conduktor/kafka-security-manager"),
+    "scm:git@github.com:conduktor/kafka-security-manager.git"
+  )
+)
+ThisBuild / developers := List(
```
Comment (simplesteph) [inline publish.sbt:9]: this doesn't seem to be in line with `build.sbt`. Is this information really needed to be duplicated?

File: publish.sbt [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+  Developer(
+    id    = "Your identifier",
+    name  = "Your Name",
+    email = "your@email",
+    url   = url("http://your.url")
+  )
+)
+
+ThisBuild / description := "Some description about your project."
```
Comment (simplesteph) [inline publish.sbt:18]: same

File: publish.sbt [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+ThisBuild / licenses := List("Apache 2" -> new URL("http://www.apache.org/licenses/LICENSE-2.0.txt"))
+ThisBuild / homepage := Some(url("https://github.com/example/project"))
```
Comment (simplesteph) [inline publish.sbt:20]: same

File: sonatype.sbt [hunk 12]
```diff
@@ -1,5 +1,5 @@
 // Your profile name of the sonatype account. The default is the same with the organization value
-sonatypeProfileName := "com.github.conduktor"
+sonatypeProfileName := "io.github.conduktor"
```
Comment (simplesteph) [inline sonatype.sbt:2]: same here

File: travis/pubring.asc [hunk 13]
```diff
@@ -0,0 +1,41 @@
+-----BEGIN PGP PUBLIC KEY BLOCK-----
```
Comment (simplesteph) [inline travis/pubring.asc:1]: why is this file under the `travis` folder? We used the use Travis but now it's github actions.. maybe make a new separate directory named differently?

File: travis/secring.asc [hunk 14]
```diff
@@ -0,0 +1,83 @@
+-----BEGIN PGP PRIVATE KEY BLOCK-----
```
Comment (simplesteph) [inline travis/secring.asc:1]: hmm....... secring.asc is a public or private file? if private, this needs to be a secret, no?

**Review by simplesteph — CHANGES_REQUESTED** [review]
Thanks for the initial PR. See comments

File: build.sbt (hunk 4 above)
Comment (Diablo2050) [reply build.sbt:2]: according to https://central.sonatype.org/publish/requirements/coordinates/ the schema is io.github.username

**Comment by Diablo2050** [comment]
Changes done

**Review by simplesteph — APPROVED** [review]
